### PR TITLE
2050: add test that ensures no packet loss during reroute

### PR DESCRIPTION
**Pull request by rtretyak**

### CODE REVIEW

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/IntentionalRerouteSpec.groovy [hunk 1]
```diff
@@ -117,7 +125,46 @@ class IntentionalRerouteSpec extends BaseSpecification {
...
+            def resources = traffExam.startExam(direction, true)
+            direction.setResources(resources)
+        }
+
+        and: "While traffic flow is active, request a flow reroute"
+        [exam.forward, exam.reverse].each { assert !traffExam.isFinished(it) }
+        northbound.rerouteFlow(flow.id)
+
+        then: "Flow is rerouted"
+        northbound.getFlowPath(flow.id) != path
```
Comment (ylobankov) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/IntentionalRerouteSpec.groovy]: Is a waiter needed here?

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/IntentionalRerouteSpec.groovy [hunk 2]
```diff
@@ -117,7 +125,46 @@ class IntentionalRerouteSpec extends BaseSpecification {
...
+
+        and: "Traffic examination result shows acceptable packet loss percentage"
+        def examReports = [exam.forward, exam.reverse].collect { traffExam.waitExam(it) }
+        examReports.each {
+            //Minor packet loss is considered a measurement error and happens regardless of reroute
+            assert it.consumerReport.lostPercent < 1
+        }
+
+        and: "Remove flow"
+        flowHelper.deleteFlow(flow.id)
```
Comment (ylobankov) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/IntentionalRerouteSpec.groovy]: It looks like we should also to delete created link props.

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/IntentionalRerouteSpec.groovy (hunk 1 above)
Comment (rtretyak) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/IntentionalRerouteSpec.groovy]: as far as I understand this operation is synchronized (actual rules installation is async, but db is updated before reroute response is returned). I didn't experience any problems here

Comment (rtretyak) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/IntentionalRerouteSpec.groovy]: moreover, the updated path is already returned as a response on L184

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/IntentionalRerouteSpec.groovy (hunk 2 above)
Comment (rtretyak) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/IntentionalRerouteSpec.groovy]: there is a `cleanup` method in this test
```
    def cleanup() {
        northbound.deleteLinkProps(northbound.getAllLinkProps())
        database.resetCosts()
    }
```

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/IntentionalRerouteSpec.groovy (hunk 1 above)
Comment (ylobankov) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/IntentionalRerouteSpec.groovy]: Ok.

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/IntentionalRerouteSpec.groovy (hunk 2 above)
Comment (ylobankov) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/IntentionalRerouteSpec.groovy]: Yep, didn't notice it.

**Pull request merged**